source/Fleet.cpp

```diff
@@ -333,7 +355,7 @@ Fleet::Variant::Variant(const DataNode &node)
 	weight = 1;
 	if(node.Token(0) == "variant" && node.Size() >= 2)
 		weight = node.Value(1);
-	else if(node.Token(0) == "add" && node.Size() >= 3)
+	else if((node.Token(0) == "add" || node.Token(0) == "remove") && node.Size() >= 3)
```

Comment:
Note that there is no reason for "remove" to specify a weight. (Unless "remove" could be used to reduce a variant's weight instead of eliminating it outright, but I think that would be way too complicated and error-prone.)